apps/src/templates/progress/ProgressBubble.jsx

```diff
@@ -1,32 +1,31 @@
 import React from 'react';
-import Radium from 'radium';
-import _ from 'lodash';
 import PropTypes from 'prop-types';
-import queryString from 'query-string';
-import i18n from '@cdo/locale';
-import color from '@cdo/apps/util/color';
-import FontAwesome from '../FontAwesome';
-import {getIconForLevel, isLevelAssessment} from './progressHelpers';
+import {isLevelAssessment} from './progressHelpers';
 import {levelWithProgressType} from './progressTypes';
 import {
-  DOT_SIZE,
-  DIAMOND_DOT_SIZE,
-  SMALL_DOT_SIZE,
-  SMALL_DIAMOND_SIZE,
-  levelProgressStyle,
-  hoverStyle
-} from './progressStyles';
-import ProgressPill from '@cdo/apps/templates/progress/ProgressPill';
-import TooltipWithIcon from './TooltipWithIcon';
+  BasicBubble,
+  BubbleLink,
+  BubbleSize,
+  BubbleTooltip,
+  getBubbleClassNames,
+  getBubbleContent,
+  getBubbleShape,
+  getBubbleUrl
+} from './BubbleFactory';
+import {levelProgressStyle} from './progressStyles';
 import {SmallAssessmentIcon} from './SmallAssessmentIcon';
-import firehoseClient from '../../lib/util/firehose';
 
-class ProgressBubble extends React.Component {
+/**
+ * A ProgressBubble represents progress for a specific level. It can be a circle
+ * or a diamond (or a pill in the case of unplugged levels), and it can be big
+ * or small. The fill and outline change depending on the level status.
+ */
+
```

Comment:
(very minor) Is this extra line here intentional?

Reply:
gross. no.